pyflagser/tests/test_flagser.py

```diff
@@ -26,9 +28,165 @@
+    },
+    'product':
+    {
+        'dgms': [
+            np.array([[0.44, 1.00999999],
+                      [0.36000001, 1.125],
+                      [0.33000001, 1.14499998],
+                      [0.88999999, 1.17999995],
+                      [0.11,        float('inf')]]),
+            np.array([[1.76999998, 2.04691648],
+                      [1.755, 1.99411869],
+                      [1.65499997, 1.97242892],
+                      [1.34000003, 1.77884996],
+                      [1.29499996, 2.08236003],
+                      [1.20000005, 2.27397013]])]
+    },
+    'sum':
+    {
+        'dgms': [
+            np.array([[0.44, 1.00999999],
+                      [0.36000001, 1.125],
+                      [0.33000001, 1.14499998],
+                      [0.88999999, 1.17999995],
+                      [0.11,        float('inf')]]),
+            np.array([[1.76999998, 3.92499995],
+                      [1.755, 3.8900001],
+                      [1.65499997, 3.84500003],
+                      [1.34000003, 3.64499998],
+                      [1.29499996, 3.83500004],
+                      [1.20000005, 4.]])]
+    },
+    'pmean':
+    {
+        'dgms': [
+            np.array([[0.44, 1.00999999],
+                      [0.36000001, 1.125],
+                      [0.33000001, 1.14499998],
+                      [0.88999999, 1.17999995],
+                      [0.11,        float('inf')]]),
+            np.array([[1.76999998, 2.54917502],
+                      [1.755, 2.52713633],
+                      [1.65499997, 2.41156578],
+                      [1.34000003, 2.04345369],
+                      [1.29499996, 2.07881474],
+                      [1.20000005, 2.43602848]])]
+    },
+    'pmoment':
+    {
+        'dgms': [
+            np.array([[0.44, 1.00999999],
+                      [0.36000001, 1.125],
+                      [0.33000001, 1.14499998],
+                      [0.88999999, 1.17999995],
+                      [0.11,        float('inf')]]),
+            np.array([[1.76999998, 1.9275918],
+                      [1.755, 1.85709834],
+                      [1.65499997, 1.7869581],
+                      [1.34000003, 1.37369251],
+                      [1.29499996, 1.39265192],
+                      [1.20000005, 1.81259179]])]
+    },
+}
+
 
 def test_flagser(flag_file):
     betti_exp = betti[os.path.split(flag_file)[1]]
     flag_matrix = loadflag(flag_file)
     betti_res = flagser(flag_matrix)['betti']
     assert_almost_equal(betti_res, betti_exp)
+
+
+def are_matrix_equal(m1, m2):
+    for i in range(min(len(m1), len(m2))):
+        m1f = np.array(m1[i]).flatten()
+        m2f = np.array(m2[i]).flatten()
+        if not np.isclose(m1f, m2f).all():
+            return False
+    return True
+
+
+def test_filtrations(flag_file):
+    """
+    Testing all filtrations available for dataset d5.flag
+    It was necessary to disable edge filtrations because of assumption that the
```

Comment:
Could you explain this in more detail (not in docstring, just here for now)? What does "disabling edge filtrations" look like in the C++ code?

Reply:
Well when testing with the data of the `d5.flag`, `vertex_degree` filtration produces a segmentation fault in C++, so I assumed it was related to the warning in the documentation about the filtration. And I copy/pasted on the docstring to explain the reason not all filtrations are tested.

But, the example I was working earlier this week, with data generated by `squareform(pdist(np.random.random((5, 2))))`. Filtration `vertex_degree` does work.


Am I wrong ?

Reply:
@MonkeyBreaker now you're talking about `vertex_degree` filtrations, but in the docstrings you are referring to "edge filtrations". Could you explain the relation?

Reply:
From the documentation:

>  filtration : string, optional, default: ``'max'``
        Algorithm determining the filtration. Warning: if an edge filtration is
        specified, it is assumed that the resulting filtration is consistent,
        meaning that the filtration value of every simplex of dimension at
        least two should evaluate to a value that is at least the maximal value
        of the filtration values of its containing edges. For performance
        reasons, this is not checked automatically.

When I was performing tests, I encountered segmentation faults running `flagser` on `d5.flag` with `vertex_degree` filtration, on the bindings and with the C++ implementation. Trying to understand why this could happen, I assumed it was related to the remark from the documentation.